commit b42ab52844123694b7f0524be08c96c8408aaa37
Author: Quentin Monnet <quentin.monnet@netronome.com>
Date:   Thu Nov 8 11:52:25 2018 +0000

    tools: bpftool: prevent infinite loop in get_fdinfo()
    
    [ Upstream commit 53909030aa29bffe1f8490df62176c2375135652 ]
    
    Function getline() returns -1 on failure to read a line, thus creating
    an infinite loop in get_fdinfo() if the key is not found. Fix it by
    calling the function only as long as we get a strictly positive return
    value.
    
    Found by copying the code for a key which is not always present...
    
    Fixes: 71bb428fe2c1 ("tools: bpf: add bpftool")
    Signed-off-by: Quentin Monnet <quentin.monnet@netronome.com>
    Reviewed-by: Jakub Kicinski <jakub.kicinski@netronome.com>
    Signed-off-by: Daniel Borkmann <daniel@iogearbox.net>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/tools/bpf/bpftool/common.c b/tools/bpf/bpftool/common.c
index b3a0709ea7ed..fcaf00621102 100644
--- a/tools/bpf/bpftool/common.c
+++ b/tools/bpf/bpftool/common.c
@@ -304,7 +304,7 @@ char *get_fdinfo(int fd, const char *key)
 		return NULL;
 	}
 
-	while ((n = getline(&line, &line_n, fdi))) {
+	while ((n = getline(&line, &line_n, fdi)) > 0) {
 		char *value;
 		int len;
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1031_b42ab52844123694b7f0524be08c96c8408aaa37.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1031_b42ab52844123694b7f0524be08c96c8408aaa37.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
